Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUBufferImpl.cpp

```diff
@@ -59,7 +59,7 @@ void BufferImpl::mapAsync(MapModeFlags mapModeFlags, Size64 offset, std::optiona
 
 auto BufferImpl::getMappedRange(Size64 offset, std::optional<Size64> size) -> MappedRange
 {
-    auto usedSize = size.value_or(WGPU_WHOLE_MAP_SIZE);
+    auto usedSize = size.value_or(wgpuBufferGetSize(m_backing));
```

Comment:
Why? WHOLE_MAP_SIZE is supposed to work.

Reply:
we can pass `WGPU_WHOLE_MAP_SIZE` to `wgpuBufferGetMappedRange` which is fine. But returning `{ pointer, static_cast<size_t>(WGPU_WHOLE_MAP_SIZE) };` is problematic, because it leads to `Ref<JSC::ArrayBuffer> GPUBuffer::getMappedRange` calling `return ArrayBuffer::create(mappedRange.source, WGPU_WHOLE_MAP_SIZE);` in GPUBuffer.cpp. Then we make a JSC::ArrayBuffer with size `WGPU_WHOLE_MAP_SIZE` which is not usable as expected from JavaScript. Calls to `.set` on the Float32Array get rejected because the sizes don't match.

`usedSize` is used in both the return statement and the call to `wgpuBufferGetMappedRange` here, but does it make sense to pass `WGPU_WHOLE_MAP_SIZE` to wgpuBufferGetMappedRange if we need the correct size anyways for the return call?